### PR TITLE
ui: remove old references to pki and old adapter

### PR DESCRIPTION
**Pull request by hellobontempo**
- Removes engine specific logic from `SecretListHeader` this should make adopting the `Hds::PageHeader` component more streamline when the time comes. 
- Removes `pki` relevant logic from general secret engine code

### CODE REVIEW

File: ui/app/routes/vault/cluster/secrets/backend/list.js [hunk 1]
```diff
@@ -144,29 +143,6 @@ export default Route.extend({
     });
   },
 
-  afterModel(model) {
```
Comment (hellobontempo) [inline ui/app/routes/vault/cluster/secrets/backend/list.js:147]: This seems only relevant to PKI, yeah?

File: ui/lib/pki/addon/components/pki-page-header.hbs
Comment (hellobontempo) [inline ui/lib/pki/addon/components/pki-page-header.hbs:1]: Making this an engine specific component allowed removing all of the `isEngine` logic from `SecretListHeader`

File: ui/lib/pki/addon/components/pki-page-header.hbs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        <li>
+          <span class="sep">/</span>
+          <LinkToExternal @route="secrets">secrets</LinkToExternal>
+        </li>
+        <li>
+          <span class="sep">/</span>
+          <span>{{@backend.id}}</span>
+        </li>
+      </ul>
+    </nav>
```
Comment (zofskeez) [inline ui/lib/pki/addon/components/pki-page-header.hbs:14]: I wonder if it's worth using the `<Page::Breadcrumbs>` component from `core` here? Not sure if that would make it easier or not when moving to HDS. Also, it seems like there is a crumb missing for the current route (overview, roles etc.).

**Review by zofskeez — COMMENTED** [review]
Really nice to see all of that conditional engine logic pulled out. I left a comment regarding the current route breadcrumb but other than that it looks good to me!

File: ui/lib/pki/addon/components/pki-page-header.hbs (hunk 2 above)
Comment (zofskeez) [reply ui/lib/pki/addon/components/pki-page-header.hbs:14]: Ahh yes. I was originally thinking about some of the child routes like configure for example but it looks like they all have their own page headers with breadcrumbs. Confused myself 😵‍💫!

**Review by zofskeez — APPROVED** [review]

**Pull request merged**